### PR TITLE
doc/radosgw: update QAT acceleration to specify the configuration

### PR DESCRIPTION
**Pull request by liumiaomiaoIntel**
Add and change the configuration steps to enable latest version of QAT driver on QAT based encryption and compression.

Signed-off-by: Miaomiao Liu <miaomiao.liu@intel.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by liumiaomiaoIntel** [comment]
@tchaikov Hi Kefu, could you review this PR which is based on PR44608?

File: doc/radosgw/qat-accel.rst [hunk 1]
```diff
@@ -69,26 +69,82 @@ hardware is available and QAT is capable to handle it.
 
 Configuration
 =============
-1. QAT based Encryption for RGW 
-
-Edit the Ceph configuration file to make use of QAT based crypto plugin::
+1. Prerequisites
+
+Make sure the QAT driver with version v1.7.L.4.14.0 or higher has
```
Comment (tchaikov) [inline doc/radosgw/qat-accel.rst:74]: wrong indent.  see https://ceph--44707.org.readthedocs.build/en/44707/radosgw/qat-accel/#configuration

File: doc/radosgw/qat-accel.rst [hunk 2]
```diff
@@ -69,26 +69,82 @@ hardware is available and QAT is capable to handle it.
...
+been installed. Remember to set an environment variable "ICP_ROOT" for your QAT
+driver package root directory. 
+
+To enable the QAT based encryption and compression, user needs to modify the QAT
+configuration files. For example, for Intel QuickAssist Adapter 8970 product, revise 
+c6xx_dev0/1/2.conf in the directory ``/etc/`` and keep them the same, e.g.::
```
Comment (tchaikov) [inline doc/radosgw/qat-accel.rst:80]: please use the `.. code-block:: ini` directive.

File: doc/radosgw/qat-accel.rst [hunk 3]
```diff
@@ -69,26 +69,82 @@ hardware is available and QAT is capable to handle it.
...
+    Dc0Name = "Dc0"
+    Dc0IsPolled = 1
+    # List of core affinities
+    Dc0CoreAffinity = 0
+
+
+2. QAT based Encryption for RGW 
+
+The CMake option ``WITH_QAT=ON`` must be configured. If you `build Ceph`_ source code,
+navigate to your cloned Ceph repository and execute the following::
```
Comment (tchaikov) [inline doc/radosgw/qat-accel.rst:108]: please use the `.. prompt:: bash $` directive.

File: doc/radosgw/qat-accel.rst [hunk 4]
```diff
@@ -69,26 +69,82 @@ hardware is available and QAT is capable to handle it.
...
+Section in QAT configuration files, e.g.::
+
+    export QAT_SECTION_NAME=CEPH
+
+Next, edit the Ceph configuration file to enable QAT support for compression::
 
     qat compressor enabled=true
 
 
+
```
Comment (tchaikov) [inline doc/radosgw/qat-accel.rst:143]: unnecessary empty line.

File: doc/radosgw/qat-accel.rst (hunk 1 above)
Comment (liumiaomiaoIntel) [reply doc/radosgw/qat-accel.rst]: Thanks for reminding. I have made some changes, and you can check it out.

File: doc/radosgw/qat-accel.rst (hunk 2 above)
Comment (liumiaomiaoIntel) [reply doc/radosgw/qat-accel.rst]: Sure, good suggestion.

File: doc/radosgw/qat-accel.rst (hunk 3 above)
Comment (liumiaomiaoIntel) [reply doc/radosgw/qat-accel.rst]: Sure, I have add this directive.

File: doc/radosgw/qat-accel.rst (hunk 4 above)
Comment (liumiaomiaoIntel) [reply doc/radosgw/qat-accel.rst]: Thanks for reminding, and I have remove the empty line.

**Comment by liumiaomiaoIntel** [comment]
@tchaikov  Hi Kefu, I have made some changes according to your suggestions. Can you please review it?

File: doc/radosgw/qat-accel.rst [hunk 5]
```diff
@@ -69,19 +69,75 @@ hardware is available and QAT is capable to handle it.
...
 Configuration
 =============
-1. QAT based Encryption for RGW 
-
-Edit the Ceph configuration file to make use of QAT based crypto plugin::
+1. Prerequisites
+
+Make sure the QAT driver with version v1.7.L.4.14.0 or higher has been installed.
+Remember to set an environment variable "ICP_ROOT" for your QAT driver package
+root directory.
```
Comment (tchaikov) [inline doc/radosgw/qat-accel.rst:76]:
```suggestion
#. Prerequisites

   Make sure the QAT driver with version v1.7.L.4.14.0 or higher has been installed.
   Remember to set an environment variable "ICP_ROOT" for your QAT driver package
   root directory. 
```

@liumiaomiaoIntel i meant to suggest something like this ^ . see https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html for more details on indent in reStructuredText.

File: doc/radosgw/qat-accel.rst [hunk 6]
```diff
@@ -69,19 +69,75 @@ hardware is available and QAT is capable to handle it.
...
+        # List of core affinities
+        Dc0CoreAffinity = 0
+
+
+2. QAT based Encryption for RGW 
+
+The CMake option ``WITH_QAT=ON`` must be configured. If you `build Ceph`_ source code,
+navigate to your cloned Ceph repository and execute the following:
+
+.. prompt:: bash $
```
Comment (tchaikov) [inline doc/radosgw/qat-accel.rst:110]: please indent properly.

File: doc/radosgw/qat-accel.rst [hunk 7]
```diff
@@ -69,19 +69,75 @@ hardware is available and QAT is capable to handle it.
...
+navigate to your cloned Ceph repository and execute the following:
+
+.. prompt:: bash $ 
+
+        cd ceph
+        ./do_cmake.sh -DWITH_QAT=ON
+        cd build
+        ininja
+
+.. note:: The section name of the QAT configuration files must be ``CEPH`` since
```
Comment (tchaikov) [inline doc/radosgw/qat-accel.rst:117]: ditto.

File: doc/radosgw/qat-accel.rst (hunk 5 above)
Comment (liumiaomiaoIntel) [reply doc/radosgw/qat-accel.rst]: Thanks for reminding. I have some changes based on your suggestion. Please review it.

File: doc/radosgw/qat-accel.rst [hunk 8]
```diff
@@ -92,3 +152,4 @@ Edit the Ceph configuration file to enable QAT support for compression::
 .. _QATzip: https://github.com/intel/QATzip
 .. _OpenSSL support for RGW encryption: https://github.com/ceph/ceph/pull/15168
 .. _QAT Engine: https://github.com/intel/QAT_Engine
+.. _build Ceph: https://docs.ceph.com/en/latest/install/build-ceph/#id1
```
Comment (tchaikov) [inline doc/radosgw/qat-accel.rst:155]: i think a more future-proof way to reference a section in this project is 

1. add a anchor at the place being referenced, something like
   ```rst
   .. _build-ceph:
   Build Ceph
   ==========
   ```
2.  and reference it like
   ```rst
   :ref:`build-ceph`
   ```

Comment (liumiaomiaoIntel) [reply doc/radosgw/qat-accel.rst]: Thanks for reminding.

**Review by tchaikov — APPROVED** [review]

**Pull request merged**